### PR TITLE
CONTEXT-1425 flatten metadata in em fit etc

### PR DESCRIPTION
**Pull request by sanderland**
This supports matchfields like "metadata.key" which is a convention used in pipelines.
Also noticed that we never released relationships update

### CODE REVIEW

File: cognite/client/_api/entity_matching.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def fit(
         capabilities in the project, are able to access the data sent to this endpoint.
 
         Args:
-            sources: entities to match from, should have an 'id' field. Tolerant to passing more than is needed or used (e.g. json dump of time series list)
+            sources: entities to match from, should have an 'id' field. Tolerant to passing more than is needed or used (e.g. json dump of time series list). Metadata fields are flattened to "metadata.key" entries for use in match_fields.
```
Comment (andreavs) [inline cognite/client/_api/entity_matching.py:144]: This kind of makes it sound like the user is supposed to do that, but the code looks like it does it for the user.

File: cognite/client/data_classes/contextualization.py [hunk 2]
```diff
@@ -214,15 +214,19 @@ def refit(self, true_matches: List[Union[Dict, Tuple[Union[int, str], Union[int,
         )
         return self._load(response.json(), cognite_client=self._cognite_client)
 
+    @staticmethod
+    def _flatten_entity(entity: Dict) -> Dict:
+        if isinstance(entity, CogniteResource):
+            entity = entity.dump(camel_case=True)
+        if "metadata" in entity:
+            for k, v in entity["metadata"].items():
+                entity["metadata.{}".format(k)] = v
```
Comment (andreavs) [inline cognite/client/data_classes/contextualization.py:223]: should there be a `del entity["metadata"]` afterwards here?

Comment (sanderland) [reply cognite/client/data_classes/contextualization.py:223]: the dict comprehension takes care of this

**Review by andreavs — DISMISSED** [review]
lgtm

File: cognite/client/_api/entity_matching.py (hunk 1 above)
Comment (sanderland) [reply cognite/client/_api/entity_matching.py]: rephrased

**Pull request merged**